components/gfx/platform/windows/font.rs

```diff
@@ -40,11 +42,205 @@ impl FontTableMethods for FontTable {
     }
 }
 
+fn make_tag(tag_bytes: &[u8]) -> FontTableTag {
+    assert_eq!(tag_bytes.len(), 4);
+    unsafe { *(tag_bytes.as_ptr() as *const FontTableTag) }
+}
+
+macro_rules! try_lossy(($result:expr) => (try!($result.map_err(|_| (())))));
+
+// Given a set of records, figure out the string indices for the family and face
+// names.  We want name_id 1 and 2, and we need to use platform_id == 1 and
+// language_id == 0 to avoid limitations in the truetype crate.  We *could* just
+// do our own parsing here, and use the offset/length data and pull the values out
+// ourselves.
+fn get_family_face_indices(records: &[truetype::naming_table::Record]) -> Option<(usize, usize)> {
+    let mut family_name_index = None;
+    let mut face_name_index = None;
+
+    for i in 0..records.len() {
+        // the truetype crate can only decode mac platform format names
+        if records[i].platform_id != 1 {
+            continue;
+        }
+
+        if records[i].language_id != 0 {
+            continue;
+        }
+
+        if records[i].name_id == 1 {
+            family_name_index = Some(i);
+        } else if records[i].name_id == 2 {
+            face_name_index = Some(i);
+        }
+    }
+
+    if family_name_index.is_some() && face_name_index.is_some() {
+        Some((family_name_index.unwrap(), face_name_index.unwrap()))
+    } else {
+        None
+    }
+}
+
+// We need the font (DWriteFont) in order to be able to query things like
+// the family name, face name, weight, etc.  On Windows 10, the
+// DWriteFontFace3 interface provides this on the FontFace, but that's only
+// available on Win10+.
+//
+// Instead, we do the parsing work using the truetype crate for raw fonts.
+// We're just extracting basic info, so this is sufficient for now.
+
+#[derive(Debug)]
+struct FontInfo {
+    family_name: String,
+    face_name: String,
+    weight: font_weight::T,
+    stretch: font_stretch::T,
+    style: FontStyle,
+}
+
+impl FontInfo {
+    fn new_from_face(face: &FontFace) -> Result<FontInfo, ()> {
+        use std::cmp::{min, max};
+        use std::io::Cursor;
+        use truetype::{NamingTable, Value, WindowsMetrics};
+
+        let name_table_bytes = face.get_font_table(make_tag(b"name"));
+        let os2_table_bytes = face.get_font_table(make_tag(b"OS/2"));
+        if name_table_bytes.is_none() || os2_table_bytes.is_none() {
+            return Err(());
+        }
+
+        let mut name_table_cursor = Cursor::new(name_table_bytes.as_ref().unwrap());
+        let names = try_lossy!(NamingTable::read(&mut name_table_cursor));
+        let (family, face) = match names {
+            NamingTable::Format0(ref table) => {
+                if let Some((family_index, face_index)) = get_family_face_indices(&table.records) {
```

Comment:
My previous comment also applied here and in the `Format1` arm, though this is not a big deal.